x-pack/plugins/osquery/public/live_queries/form/index.tsx

```diff
@@ -399,70 +392,38 @@ const LiveQueryFormComponent: React.FC<LiveQueryFormProps> = ({
     [actionId, agentIds, data?.actions, hideFullscreen]
   );
 
-  const formSteps: EuiContainedStepProps[] = useMemo(
-    () => [
-      {
-        title: i18n.translate('xpack.osquery.liveQueryForm.steps.agentsStepHeading', {
-          defaultMessage: 'Select agents',
-        }),
-        children: <UseField path="agentSelection" component={AgentsTableField} />,
-        status: agentSelected ? 'complete' : 'incomplete',
-      },
-      {
-        title: i18n.translate('xpack.osquery.liveQueryForm.steps.queryStepHeading', {
-          defaultMessage: 'Enter query',
-        }),
-        children: queryFieldStepContent,
-        status: queryStatus,
-      },
-      {
-        title: i18n.translate('xpack.osquery.liveQueryForm.steps.resultsStepHeading', {
-          defaultMessage: 'Check results',
-        }),
-        children: resultsStepContent,
-        status: resultsStatus,
-      },
-    ],
-    [agentSelected, queryFieldStepContent, queryStatus, resultsStepContent, resultsStatus]
-  );
-
-  const simpleForm = useMemo(
-    () => (
-      <EuiFlexGroup direction="column">
-        <UseField
-          path="agentSelection"
-          component={agentsField ? AgentsTableField : GhostFormField}
-        />
-        <EuiFlexItem>{queryFieldStepContent}</EuiFlexItem>
-        <EuiFlexItem>{resultsStepContent}</EuiFlexItem>
-      </EuiFlexGroup>
-    ),
-    [agentsField, queryFieldStepContent, resultsStepContent]
-  );
-
   useEffect(() => {
-    if (defaultValue?.agentSelection) {
-      setFieldValue('agentSelection', defaultValue?.agentSelection);
-    }
-
-    if (defaultValue?.query) {
-      setFieldValue('query', defaultValue?.query);
-    }
-
-    // TODO: Set query and ECS mapping from savedQueryId object
-    if (defaultValue?.savedQueryId) {
-      setFieldValue('savedQueryId', defaultValue?.savedQueryId);
-    }
-
-    if (!isEmpty(defaultValue?.ecs_mapping)) {
-      setFieldValue('ecs_mapping', defaultValue?.ecs_mapping);
+    if (defaultValue) {
+      updateFieldValues({
+        agentSelection: defaultValue.agentSelection,
+        query: defaultValue.query,
+        savedQueryId: defaultValue.savedQueryId,
+        ecs_mapping: defaultValue.ecs_mapping
+          ? map(defaultValue.ecs_mapping, (value, key) => ({
+              key,
+              result: {
+                type: Object.keys(value)[0],
+                value: Object.values(value)[0],
+              },
+            }))
+          : undefined,
+      });
     }
-  }, [defaultValue, setFieldValue]);
+  }, [defaultValue, updateFieldValues]);
 
   return (
     <>
       <Form form={form}>
-        {formType === 'steps' ? <EuiSteps steps={formSteps} /> : simpleForm}
+        <EuiFlexGroup direction="column">
+          <EuiFlexItem>
+            <UseField
+              path="agentSelection"
+              component={!hideAgentsField ? AgentsTableField : GhostFormField}
```

Comment:
why do we hide agents field? (it's just for osquery action right?)

Reply:
correct